Azure Eventhub receiver stops consuming after an unmarshal error

### Component(s)

receiver/azureeventhub

### What happened?

## Description
Every time the receiver logs the following message, it stops consuming events (immediately or after a couple of minutes)

2026-02-05T10:04:07.230Z error azureeventhubreceiver@v0.144.0/eventhubhandler.go:117 Error reported by event hub {"resource": {"service.instance.id": "6d025f31-06f0-419c-a92b-b7438a97c7b4", "service.name": "otelcol-contrib", "service.version": "0.144.0"}, "otelcol.component.id": "azureeventhub", "otelcol.component.kind": "receiver", "otelcol.signal": "traces", "error": "failed to unmarshal traces: expected { character for map value"}
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver.(*eventhubHandler).setUpOnePartition.func1
 github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver@v0.144.0/eventhubhandler.go:117

Restarting the collector will resume events consumption, until the next parsing error.

## Steps to Reproduce

Configure App Insights to stream AppTraces/Dependencies/Requests to an eventhub

Consume the events using the receiver

Eventually it will fail to parse a message and stop working completely


## Expected Result
Parsing/Unmarshal errors should not break the receiver and put it in an unrecoverable state

## Actual Result
Receiver stops working completely, no logs, no activity, no retries, nothing

### Collector version

0.144.0

### Environment information

## Environment
Standard_E8s vms running on Ubuntu


### OpenTelemetry Collector configuration

```yaml

```

### Log output

```shell
2026-02-05T10:04:07.230Z error azureeventhubreceiver@v0.144.0/eventhubhandler.go:117 Error reported by event hub {"resource": {"service.instance.id": "6d025f31-06f0-419c-a92b-b7438a97c7b4", "service.name": "otelcol-contrib", "service.version": "0.144.0"}, "otelcol.component.id": "azureeventhub", "otelcol.component.kind": "receiver", "otelcol.signal": "traces", "error": "failed to unmarshal traces: expected { character for map value"}
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver.(*eventhubHandler).setUpOnePartition.func1
 github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver@v0.144.0/eventhubhandler.go:117
```

### Additional context

_No response_

### Tip

<sub>[React](https://github.blog/news-insights/product-news/add-reactions-to-pull-requests-issues-and-comments/) with 👍 to help prioritize this issue. Please use comments to provide useful context, avoiding `+1` or `me too`, to help us triage it. Learn more [here](https://opentelemetry.io/community/end-user/issue-participation/).</sub>

--- a/.chloggen/azure-eventhub-continue-after-error.yaml
+++ b/.chloggen/azure-eventhub-continue-after-error.yaml
@@ -4,7 +4,7 @@
 change_type: bug_fix
 
 # The name of the component, or a single word describing the area of concern, (e.g. receiver/filelog)
-component: receiver/azureeventhub
+component: receiver/azure_event_hub
 
 # A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
 note: Fixes a bug where the receiver would stop receiving messages after a parsing error.

--- a/.chloggen/azure-eventhub-continue-after-error.yaml
+++ b/.chloggen/azure-eventhub-continue-after-error.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. receiver/filelog)
+component: receiver/azureeventhub
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fixes a bug where the receiver would stop receiving messages after a parsing error.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [45898]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: []

--- a/receiver/azureeventhubreceiver/eventhubhandler.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler.go
@@ -18,7 +18,7 @@ import (
 
 type hubWrapper interface {
 	GetRuntimeInformation(ctx context.Context) (*hubRuntimeInfo, error)
-	Receive(ctx context.Context, partitionID string, handler hubHandler, applyOffset bool) (listenerHandleWrapper, error)
+	Receive(ctx context.Context, partitionID string, handler hubHandler, applyOffset bool, logger *zap.Logger) (listenerHandleWrapper, error)
 	Close(ctx context.Context) error
 }
 
@@ -98,7 +98,7 @@ func (h *eventhubHandler) run(ctx context.Context, host component.Host) error {
 }
 
 func (h *eventhubHandler) setUpOnePartition(ctx context.Context, partitionID string, applyOffset bool) error {
-	handle, err := h.hub.Receive(ctx, partitionID, h.newMessageHandler, applyOffset)
+	handle, err := h.hub.Receive(ctx, partitionID, h.newMessageHandler, applyOffset, h.settings.Logger)
 	if err != nil {
 		return err
 	}

--- a/receiver/azureeventhubreceiver/eventhubhandler_azeventhub.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler_azeventhub.go
@@ -172,7 +172,7 @@ func (h *hubWrapperAzeventhubImpl) GetRuntimeInformation(ctx context.Context) (*
 	return nil, errNoConfig
 }
 
-func (h *hubWrapperAzeventhubImpl) Receive(ctx context.Context, partitionID string, handler hubHandler, applyOffset bool) (listenerHandleWrapper, error) {
+func (h *hubWrapperAzeventhubImpl) Receive(ctx context.Context, partitionID string, handler hubHandler, applyOffset bool, logger *zap.Logger) (listenerHandleWrapper, error) {
 	if h.hub != nil {
 		namespace, err := h.namespace()
 		if err != nil {
@@ -223,16 +223,16 @@ func (h *hubWrapperAzeventhubImpl) Receive(ctx context.Context, partitionID stri
 				events, err := pc.ReceiveEvents(timeout, maxPollEvents, nil)
 				cancelTimeout()
 				if err != nil && !errors.Is(err, context.DeadlineExceeded) {
-					w.setErr(err)
-					return
+					logger.Error("error receiving events", zap.Error(err))
+					continue
 				}
 
 				for _, ev := range events {
 					if err := handler(ctx, &azureEvent{
 						AzEventData: ev,
 					}); err != nil {
-						w.setErr(err)
-						return
+						logger.Error("error processing event", zap.Error(err))
+						continue
 					}
 				}
 
@@ -246,8 +246,8 @@ func (h *hubWrapperAzeventhubImpl) Receive(ctx context.Context, partitionID stri
 							},
 						)
 						if err != nil {
-							w.setErr(err)
-							return
+							logger.Error("error writing checkpoint", zap.Error(err))
+							continue
 						}
 					}
 				}

--- a/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
@@ -25,9 +25,16 @@ import (
 type mockPartitionClient struct {
 	eventData []*azeventhubs.ReceivedEventData
 	closed    bool
+	err       error
+	callCount int
 }
 
 func (p *mockPartitionClient) ReceiveEvents(_ context.Context, maxBatchSize int, _ *azeventhubs.ReceiveEventsOptions) ([]*azeventhubs.ReceivedEventData, error) {
+	p.callCount++
+	if p.err != nil && p.callCount == 1 {
+		return nil, p.err
+	}
+
 	events := make([]*azeventhubs.ReceivedEventData, 0, maxBatchSize)
 
 	for i := range maxBatchSize {
@@ -54,6 +61,7 @@ type mockAzeventHub struct {
 	partitionID         string
 	offset              string
 	closed              bool
+	partitionClient     *mockPartitionClient
 }
 
 func (a *mockAzeventHub) GetEventHubProperties(_ context.Context, _ *azeventhubs.GetEventHubPropertiesOptions) (azeventhubs.EventHubProperties, error) {
@@ -69,7 +77,9 @@ func (a *mockAzeventHub) NewPartitionClient(partitionID string, options *azevent
 	if options != nil && options.StartPosition.Offset != nil {
 		a.offset = *options.StartPosition.Offset
 	}
-
+	if a.partitionClient != nil {
+		return a.partitionClient, nil
+	}
 	return &mockPartitionClient{}, nil
 }
 
@@ -262,6 +272,22 @@ func TestPartitionListener_SetErr(t *testing.T) {
 	require.NoError(t, p.err)
 	p.setErr(errors.New("test"))
 	assert.Equal(t, "test", p.err.Error())
+}
+
+func TestReceive_ContinuesAfterError(t *testing.T) {
+	pc := &mockPartitionClient{err: errors.New("receive error")}
+	h := &hubWrapperAzeventhubImpl{
+		hub:    &mockAzeventHub{partitionClient: pc},
+		config: &Config{Connection: "Endpoint=sb://test.servicebus.windows.net/;SharedAccessKeyName=Key;SharedAccessKey=Secret;EntityPath=hub", PollRate: 1},
+	}
+
+	ctx, cancel := context.WithTimeout(t.Context(), 2*time.Second)
+	defer cancel()
+
+	_, err := h.Receive(ctx, "p1", func(_ context.Context, _ *azureEvent) error { return nil }, false, zaptest.NewLogger(t))
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool { return pc.callCount >= 2 }, 2*time.Second, 50*time.Millisecond)
 }
 
 func TestGetConsumerGroup(t *testing.T) {

--- a/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
@@ -7,6 +7,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -26,12 +27,12 @@ type mockPartitionClient struct {
 	eventData []*azeventhubs.ReceivedEventData
 	closed    bool
 	err       error
-	callCount int
+	callCount atomic.Int32
 }
 
 func (p *mockPartitionClient) ReceiveEvents(_ context.Context, maxBatchSize int, _ *azeventhubs.ReceiveEventsOptions) ([]*azeventhubs.ReceivedEventData, error) {
-	p.callCount++
-	if p.err != nil && p.callCount == 1 {
+	callCount := p.callCount.Add(1)
+	if p.err != nil && callCount == 1 {
 		return nil, p.err
 	}
 
@@ -287,7 +288,7 @@ func TestReceive_ContinuesAfterError(t *testing.T) {
 	_, err := h.Receive(ctx, "p1", func(_ context.Context, _ *azureEvent) error { return nil }, false, zaptest.NewLogger(t))
 	require.NoError(t, err)
 
-	require.Eventually(t, func() bool { return pc.callCount >= 2 }, 2*time.Second, 50*time.Millisecond)
+	require.Eventually(t, func() bool { return pc.callCount.Load() >= 2 }, 2*time.Second, 50*time.Millisecond)
 }
 
 func TestGetConsumerGroup(t *testing.T) {

--- a/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler_azeventhub_test.go
@@ -19,6 +19,7 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
 )
 
 type mockPartitionClient struct {
@@ -171,6 +172,7 @@ func TestHubWrapperAzeventhubImpl_Receive(t *testing.T) {
 					return nil
 				},
 				test.applyOffset,
+				zaptest.NewLogger(t),
 			)
 			if test.expectErr {
 				require.Error(t, err)

--- a/receiver/azureeventhubreceiver/eventhubhandler_test.go
+++ b/receiver/azureeventhubreceiver/eventhubhandler_test.go
@@ -35,7 +35,7 @@ func (mockHubWrapper) GetRuntimeInformation(context.Context) (*hubRuntimeInfo, e
 	}, nil
 }
 
-func (mockHubWrapper) Receive(ctx context.Context, _ string, _ hubHandler, _ bool) (listenerHandleWrapper, error) {
+func (mockHubWrapper) Receive(ctx context.Context, _ string, _ hubHandler, _ bool, _ *zap.Logger) (listenerHandleWrapper, error) {
 	return &mockListenerHandleWrapper{
 		ctx: ctx,
 	}, nil

